src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,99 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import './TodoList.scss';
+import classNames from 'classnames';
 
-export const TodoList = () => (
+export const TodoList = ({
+  todos,
+  onUserSelection,
+  onCompletedChange,
+  userId,
+  inputValue,
+  onTodosFilter,
+  selectValue,
+  onSelectValueChange,
+}) => (
   <div className="TodoList">
-    <h2>Todos:</h2>
+    <h2>{`Todos: ${todos.length}`}</h2>
+
+    <div>
+      <input
+        placeholder="Search todos"
+        type="text"
+        value={inputValue}
+        onChange={onTodosFilter}
+      />
+
+      <select
+        name=""
+        id=""
+        value={selectValue}
+        onChange={onSelectValueChange}
+      >
+        <option value="all">
+          All
+        </option>
+        <option value="active">
+          Active
+        </option>
+        <option value="completed">
+          Completed
+        </option>
+      </select>
+    </div>
 
     <div className="TodoList__list-container">
       <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
+        {todos.map(todo => (
+          <li
+            key={todo.id}
+            className={classNames('TodoList__item', {
+              'TodoList__item--checked': todo.completed,
+              'TodoList__item--unchecked': !todo.completed,
+            })}
           >
-            User&nbsp;#1
-          </button>
-        </li>
+            <label>
+              <input
+                type="checkbox"
+                checked={todo.completed}
+                onChange={() => {
+                  onCompletedChange(todo.id);
+                }}
+                readOnly
+              />
+              <p>{todo.title}</p>
+            </label>
 
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
+            <button
+              className={classNames('TodoList__user-button', 'button', {
+                'TodoList__user-button--selected': todo.userId === userId,
```

Comment:
this doesn't work as expected, all buttons have this classname no matter what user is selected

Reply:
I did this intentionally, I thought that when I select the user it activates the buttons on all of the todos of this user. 
The buttons of not selected the user aren't changed.
https://www.loom.com/share/565484eb539f41a4a885d6c4e89e79e6

Reply:
ok, got it